google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -245,12 +257,48 @@ public Blob get(BlobId blob) {
 
   @Override
   public Page<Bucket> list(BucketListOption... options) {
-    return todo();
+    UnaryCallable<ListBucketsRequest, ListBucketsPagedResponse>
+        listBucketsRequestListBucketsPagedResponseUnaryCallable =
+            grpcStorageStub.listBucketsPagedCallable();
+    final Map<StorageRpc.Option, ?> optionsMap = StorageImpl.optionMap(options);
+    ListBucketsRequest.Builder builder = ListBucketsRequest.newBuilder();
+    ifNonNull(
+        (Long) optionsMap.get(StorageRpc.Option.MAX_RESULTS), Long::intValue, builder::setPageSize);
+    ifNonNull((String) optionsMap.get(StorageRpc.Option.PAGE_TOKEN), builder::setPageToken);
+    ifNonNull((String) optionsMap.get(StorageRpc.Option.PREFIX), builder::setPrefix);
+    // TODO(sydmunro): StorageRpc.Option.Fields
+    // TODO(sydmunro): User Project
```

Comment:
According to the protos, project is required to be able to list buckets.

We'll probably need to do a similar thing here as for listHmacKeys with the fallback to options.

Reply:
User project is different though because it is used for requester pays to refer to the billing project, correct? the required field that you are referring to is the projectid right?

Reply:
+1 to @sydney-munro, you're right, UserProject is meant to assign operation cost to a specified project. It's usually used with RequesterPays but may still be used even if requesterPays is not enabled.

Reply:
According to https://github.com/googleapis/java-storage/blob/8a7755cc8352b3ab21c252885fb86576474d7f09/proto-google-cloud-storage-v2/src/main/proto/google/storage/v2/storage.proto#L336 a project must be specified in a list buckets request.

If that is incorrect, we should request that the protos be updated.

Reply:
yeah, what I am referring to in this comment is that 
https://github.com/googleapis/java-storage/blob/main/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java#L1015-L1017
is an option but not supported by the proto. The project parent field for ht ListBucketsRequest is the project id which is something else and should be fixed now with https://github.com/googleapis/java-storage/pull/1439/files#diff-8d6230c7cebeacb950a63216b0068065a15d4c2943ac7f18405906a38b3d5f66R263

Reply:
Ahh, missed that in the diff-diffs.